### PR TITLE
QuoteCreator - Fixes #20

### LINKED ISSUE #20: Random Quote Area

### PR DESCRIPTION
**Pull request by S1robe**
# Description

Adds a section to welcome page that will draw random quotes for the user on open, and per day.
Loads from rootBundle's assets to parse the quotes as YamlLists.

Fixes #20

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

See tests/decorations_test.dart: 
  - The test renders the widget, then checks if the text changes internally after being pressed.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Benek6h — APPROVED** [review]
Looks good. We'll handle overlapping tests and unused code in another pull request.

File: app/assets/quotes.yml
Comment (JikoJeans) [inline app/assets/quotes.yml:1]: Awesome this file is easily accessible and easy to populate.

File: app/lib/main.dart
Comment (JikoJeans) [inline app/lib/main.dart:1]: Thank you for correcting the boot up page of the app and for creating a function that can be used to load resources before displaying the welcome page,  its very useful for future features.

**Review by Cooljawty — APPROVED** [review]
Every thing looks good here.
Seperating the quote widget from the welcome page was also a good choice.

File: app/lib/uiwidgets/decorations.dart [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  State<Quote> createState() => _QuoteState();
+}
+
+class _QuoteState extends State<Quote> {
+  late String currentQuote;
+
+  @override
+  void initState() {
+    super.initState();
+    assert(quotes != null, throw StateError("Quotes were not properly initialized. Cannot continue."));
```
Comment (JikoJeans) [inline app/lib/uiwidgets/decorations.dart:25]: Good catch considering if the quotes are not fully loaded. Its best to consider these cases when working with resources and this check should be used moving forward on any resources called for other features

**Pull request merged**